src/template-analyzer.ts

```diff
@@ -198,19 +198,40 @@ export class TemplateAnalyzer {
    */
   public resolveLocation(
     loc: parse5.Location
-  ): ESTree.SourceLocation | null | undefined {
+  ): ESTree.SourceLocation {
     let offset = 0;
+    let expression;
+    let height = 0;
 
     for (const quasi of this._node.quasi.quasis) {
       const placeholder = getExpressionPlaceholder(this._node, quasi);
       offset += quasi.value.raw.length + placeholder.length;
 
+      const i = this._node.quasi.quasis.indexOf(quasi);
+      if (i !== 0) {
+        expression = this._node.quasi.expressions[i - 1];
```

Comment:
i think this can be scoped now as you only use it here. as in:

```ts
const expression \ this._node.quasi.expressions[i - 1];
```

and get rid of the `let expression;` further up

Reply:
fixed 👍